### ISSUE TITLE
"failed to get certificate URL" error

### ISSUE THREAD

**Issue by giuliov** [Microsoft/vsts-image-generation#27]
I tried to use it but I always get similar errors
```
ERROR: -> Forbidden : Access denied
 ...failed to get certificate URL
Build 'managed-disk' errored: Failed to fetch secret from pkrkvtyzng9ky7m/packerKeyVaultSecret, HTTP status code=403 (Forbidden)
```

The application account I use is quite powerful (I use it to create and tear-down resources via Terraform) and the KeyVault is created by Packer itself, so I am lost at what I am doing wrong.

**Comment by lokesh755** [comment]
Can you try removing "managed-disk" builder from the json and try again to see if you get this error again ?

**Comment by giuliov** [comment]
Same error, this time from `vhd` provider.

**Comment by chrisrpatterson** [comment]
@giuliov you are likely missing the tenentid from your variables.  Also the issue you are facing is a general packer error and not specific to this repo.

**Comment by DeChrist** [comment]
tl;dr you can get the correct object ID to use via the PowerShell cmdlet Get-AzureRmADServicePrincipal

@giuliov This occurred to me when I specified an incorrect ObjectId in the variables.
This packer repo Issue describes this in more details: https://github.com/hashicorp/packer/issues/4201
